### PR TITLE
Have psalm analysis directly on github

### PR DESCRIPTION
**Pull request by rullzer**
Lets see if this helps giving us some cool psalm warnings

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]
:+1: :elephant:

**Comment by LukasReschke** [comment]
This seems to fail at 97%? https://github.com/nextcloud/server/pull/24904/checks?check_run_id=1866463414

**Comment by LukasReschke** [comment]
Oh wait. I think I have an idea why, let me play around with it :)

**Comment by LukasReschke** [comment]
So the run itself will fail as Psalm will `exit(2)` here due to https://github.com/vimeo/psalm/blob/2dc86651a411df6c673079b37df20103705bb4d2/src/Psalm/IssueBuffer.php#L684-L689. Only for the Security Analysis it won't do an exit if results exist.

We should add `continue-on-error: true` to this. This _may_ then work. Let me give that a try on your branch.

**Comment by LukasReschke** [comment]
@rullzer  see https://github.com/nextcloud/server/security/code-scanning?query=ref%3Arefs%2Fpull%2F24904%2Fmerge+tag%3Amaintainability 

It also has added the tag `maintainability`. So that is distinct from the `security` one.

File: .github/workflows/psalm-github.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Psalm show github
+
+on:
+  pull_request:
```
Comment (LukasReschke) [inline .github/workflows/psalm-github.yml:4]: I'd propose also to add `push` or the comparison to the base branch may fail.

```suggestion
on:
  push:
  pull_request:
```

**Comment by rullzer** [comment]
Maybve we should also run a daily scan. Just so that the overview keeps updated

**Comment by rullzer** [comment]
Ok all good I think!

**Review by LukasReschke — APPROVED** [review]

**Comment by ChristophWurst** [comment]
> **Code scanning results / Psalm ** Completed in 4s — 1 analysis not found

expected?

**Comment by LukasReschke** [comment]
@ChristophWurst Yep. That’s the base comparison that I spoke earlier about. We need to run this on master continuously  :)

**Pull request merged**